### PR TITLE
Adds SQS Sender implementation using AWS V2 SDK

### PR DESCRIPTION
**Pull request by llinder**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
   <modules>
     <module>sender-sqs</module>
+    <module>sender-sqs2</module>
```
Comment (adriancole) [inline pom.xml]: how about `sender-awssdk-sqs`? I was recently looking around for a way to make sense of the different versions and not get into the marketing vs artifact wars. If you plop this search it surprisingly works https://search.maven.org/search?q=awssdk-sqs

Comment (llinder) [reply pom.xml]: I like that better.  thanks for the suggestion!

File: sender-sqs2/src/main/java/zipkin2/reporter/sqs2/SQSSender2.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      if (encodedSpans[i] >= 0x80) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public final String toString() {
+    return "SQSSender2{queueUrl= " + queueUrl + "}";
```
Comment (adriancole) [inline sender-sqs2/src/main/java/zipkin2/reporter/sqs2/SQSSender2.java]: I would change the package name, but not the type name. Remember Kafka 0.10. wait.. 0.11.. crap 1.0.. aw man 2.3 :)

**Comment by llinder** [comment]
Package and artifact names updated.  I also snuck in an asynchronous variant of the sender.  I suspect that projects already using the V2 SDK will likely use the netty async client unless they have specific needs to avoid netty, in which case they can use the synchronous client instead and optionally remove the apache http client in favor of the urlconnection client if desired.

**Pull request merged**